tracing: add cluster settting controlling span creation

andreimatei opened the pull request:
This patch adds the trace.span_registry.enable cluster setting to
controls whether tracing spans are created and registered with the
active spans registry. The cluster setting defaults to true, just like
the env var before it. The point of the patch is to allow one to stop
span creation at runtime if anything goes awry.

Release note: None
Release justification: low-risk updates to new functionality

andreimatei committed:
tracing: add cluster settting controlling span creation

This patch adds the trace.span_registry.enable cluster setting to
controls whether tracing spans are created and registered with the
active spans registry. The cluster setting defaults to true, just like
the env var before it. The point of the patch is to allow one to stop
span creation at runtime if anything goes awry.

Release note: None
Release justification: low-risk updates to new functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/77303)
<!-- Reviewable:end -->

andreimatei committed:
*: minor Tracer construction cleanups

A few tests were reaching directly into their Tracer to configure it.
There's now a better way.

Release note: None
Release justification: non-production code changes

abarganier reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> sans a stale comment

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77303#-:-MxBchy7CXycjyfKR_Xy:b-sfdsg5)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier and @andreimatei)

___
*[pkg/util/tracing/tracer.go, line 533 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/77303#-MxBar1z8S5xFVGJ2Nxs:-MxBar1z8S5xFVGJ2Nxt:b-r1c1qn) ([raw file](https://github.com/cockroachdb/cockroach/blob/ec214268b1f732b003b0fe4f731f99262aabd689/pkg/util/tracing/tracer.go#L533)):*
> ```Go
> 
> // NewTracer creates a Tracer. It initially tries to run with minimal overhead
> // and collects essentially nothing; use Configure() to enable various tracing
> ```

nit: The bit about collecting essentially nothing has gone stale 



_Code quote:_
```Go
// NewTracer creates a Tracer. It initially tries to run with minimal overhead
// and collects essentially nothing; use Configure() to enable various tracing
// backends.
```


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77303)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier)

___
*[pkg/util/tracing/tracer.go, line 533 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/77303#-MxBar1z8S5xFVGJ2Nxs:-MxBfWfhCo-_184E_Svz:b-lhs69o) ([raw file](https://github.com/cockroachdb/cockroach/blob/ec214268b1f732b003b0fe4f731f99262aabd689/pkg/util/tracing/tracer.go#L533)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

nit: The bit about collecting essentially nothing has gone stale 
</blockquote></details>

done, thanks


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/77303#-:-MxGaEYr0ceCArE22Q0c:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @abarganier)



<!-- Sent from Reviewable.io -->
